iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/metadata/TableMetadataImpl.java

```diff
@@ -188,6 +193,278 @@ && isIntegerNumber(argumentTypes.get(2)))) {
+                + " only accepts one argument and it must be TimeStamp, Double, Float, Int32 or Int64 data type.");
+      }
+      return DOUBLE;
+    } else if (TableBuiltinScalarFunction.COSH.getFunctionName().equalsIgnoreCase(functionName)) {
+      if (!(argumentTypes.size() == 1 && isNumericType(argumentTypes.get(0)))) {
+        throw new SemanticException(
+            "Scalar function "
+                + functionName.toLowerCase(Locale.ENGLISH)
+                + " only accepts one argument and it must be TimeStamp, Double, Float, Int32 or Int64 data type.");
+      }
+      return DOUBLE;
+    } else if (TableBuiltinScalarFunction.TANH.getFunctionName().equalsIgnoreCase(functionName)) {
+      if (!(argumentTypes.size() == 1 && isNumericType(argumentTypes.get(0)))) {
+        throw new SemanticException(
+            "Scalar function "
+                + functionName.toLowerCase(Locale.ENGLISH)
+                + " only accepts one argument and it must be TimeStamp, Double, Float, Int32 or Int64 data type.");
+      }
+      return DOUBLE;
+    } else if (TableBuiltinScalarFunction.DEGREES
+        .getFunctionName()
+        .equalsIgnoreCase(functionName)) {
+      if (!(argumentTypes.size() == 1 && isNumericType(argumentTypes.get(0)))) {
+        throw new SemanticException(
+            "Scalar function "
+                + functionName.toLowerCase(Locale.ENGLISH)
+                + " only accepts one argument and it must be TimeStamp, Double, Float, Int32 or Int64 data type.");
+      }
+      return DOUBLE;
+    } else if (TableBuiltinScalarFunction.RADIANS
+        .getFunctionName()
+        .equalsIgnoreCase(functionName)) {
+      if (!(argumentTypes.size() == 1 && isNumericType(argumentTypes.get(0)))) {
+        throw new SemanticException(
+            "Scalar function "
+                + functionName.toLowerCase(Locale.ENGLISH)
+                + " only accepts one argument and it must be TimeStamp, Double, Float, Int32 or Int64 data type.");
+      }
+      return DOUBLE;
+    } else if (TableBuiltinScalarFunction.ABS.getFunctionName().equalsIgnoreCase(functionName)) {
+      if (!(argumentTypes.size() == 1 && isNumericType(argumentTypes.get(0)))) {
+        throw new SemanticException(
+            "Scalar function "
+                + functionName.toLowerCase(Locale.ENGLISH)
+                + " only accepts one argument and it must be TimeStamp, Double, Float, Int32 or Int64 data type.");
+      }
+      return argumentTypes.get(0);
+    } else if (TableBuiltinScalarFunction.SIGN.getFunctionName().equalsIgnoreCase(functionName)) {
+      if (!(argumentTypes.size() == 1 && isNumericType(argumentTypes.get(0)))) {
+        throw new SemanticException(
+            "Scalar function "
+                + functionName.toLowerCase(Locale.ENGLISH)
+                + " only accepts one argument and it must be TimeStamp, Double, Float, Int32 or Int64 data type.");
+      }
+      return INT32;
```

Comment:
return type should same as input(https://trino.io/docs/current/functions/math.html#sign)
<img width="479" alt="image" src="https://github.com/user-attachments/assets/fe9f55ff-4a36-4f2d-ab48-9edcb6a033b8">

return argumentTypes.get(0);

Reply:
for timestamp dataType, its return type should also be long